src/menus/menus.service.ts

```diff
@@ -160,30 +160,67 @@ export class MenusService {
   }
 
   async updateOne(id: string, menuData: CreateMenuDto) {
-    await this.menuModel.updateOne({ _id: id }, menuData).exec();
+    const oldMenu = await this.menuModel.findByIdAndUpdate(id, menuData).exec();
+    if (oldMenu.category._id === menuData.category) {
+      return;
+    }
+
+    await this.categoriesService.pullMenuFromCategory(
+      oldMenu.category._id,
+      oldMenu._id,
+    );
+
+    if (!menuData.category) {
```

Comment:
if the category is `null` don't we need to put it in `uncategorized` category?

Reply:
Gotcha.